opal/mca/btl/openib/btl_openib_mca.c

```diff
@@ -641,6 +641,9 @@ int btl_openib_register_mca_params(void)
         /* Don't try to recover from this */
         return OPAL_ERR_OUT_OF_RESOURCE;
     }
+    if (NULL != mca_btl_openib_component.default_recv_qps) {
+        free(mca_btl_openib_component.default_recv_qps);
+    }
     mca_btl_openib_component.default_recv_qps = default_qps;
```

Comment:
Yeah, I think it was mostly a side effect of a conservative style, and to help with debugging (i.e., if we *never* call `free()` (and friends) with a NULL pointer, then if OPAL intercepts ever saw `free()` with NULL, it was a bug).